src/geo/ui/legends/bubble/legend-view.js

```diff
@@ -23,7 +23,7 @@ var BubbleLegendView = LegendViewBase.extend({
 
   _getCompiledTemplate: function () {
     return template({
-      labels: this.model.get('values').slice(0).reverse(),
+      labels: this._reverseIfNeeded(this.model).values,
```

Comment:
To make it a bit more consistent, I'd call this `this._calculateLabels`, and do something like (`_reverseIfNeeded` is not expressing a clear intention IMO):

```
calculateLabels: function () {
  var labels = this.model.get('values').slice(0);
  if (this._areSizesInAscendingOrder()) {
    labels = labels.reverse();
  }
  return labels;
},

_calculateBubbleSizes: function () {
  var sizes = this.model.get('sizes').slice(0).reverse()
  if (this._areSizesInAscendingOrder()) {
    sizes = sizes.revers();
  }
},

_areSizesInAscendingOrder: function () {
  var sizes = this.model.get('sizes').slice(0);
  return _.first(sizes) < _.last(sizes);
},
...
```

Just an 💡 .